### DISCUSSION TITLE
Can't specify parent directory for untitled resource

### DISCUSSION THREAD

**Discussion by meisenbarth-work** [eclipse-theia/theia#13159] [Q&A] [answered]
I tried to create a new document using UntitledResourceResolver.createUntitledURI and pass a parent, but the result always is untitled:///Untitled-1.

Checking node_modules\@theia\core\src\common\resource.ts reveals it is overwritten in 367 and the result from the parent (365) is not used. Is this intended behavior or is a else missing?

**Comment by msujew** [comment]
I believe this is supposed to be intended. Note that untitled files don't really exist on the file system (yet), so there is little need to have a parent directory. What is your use case that requires one?

**Reply by meisenbarth-work** [reply]
When i try to save it, Theia suggest my root drive as location instead of the workspace, I assume that is the case since the file is /Untitled-1.

**Reply by msujew** [reply, marked as answer]
I see, yeah that's unrelated. So the issue itself isn't that you're unable to specify the parent directory, but rather that the default saving mechanism doesn't respect the current workspace when saving untitled resources. It's likely we need an improvement in our [FilesystemSaveResourceService](https://github.com/eclipse-theia/theia/blob/master/packages/filesystem/src/browser/filesystem-save-resource-service.ts).

**Reply by meisenbarth-work** [reply]
Maybe there are two solutions to the problem. When I change the uri after UntitledResourceResolver.createUntitledURI before passing it to this.UntitledResourceResolver.createUntitledResource Theia suggest the expected directory when I save. But since this is the expected behavior i will rebind UntitledResourceResolver and overwrite createUntitledURI to contain an else around [line 367](https://github.com/eclipse-theia/theia/blob/master/packages/core/src/common/resource.ts#L367).

**Reply by meisenbarth-work** [reply]
Compared with VS Code your solution is probably the correct one, since VS Code suggest the folder with the most recently viewed file when saving and not the current location from when creating the file.
